src/js/entities-service/entities/states.js

```diff
@@ -17,6 +18,15 @@ const Model = Store(_Model, TYPE);
 const Collection = BaseCollection.extend({
   url: '/api/states',
   model: Model,
+  filterByDone(onlyIncludeDone = false) {
+    const filteredStates = this.filter(state => {
+      if (onlyIncludeDone) return state.isDone();
+
+      return !state.isDone();
+    });
+
+    return new Backbone.Collection(filteredStates);
+  },
```

Comment:
Returning a new backbone collection instead of modifying `this`.

In the `worklist_state.js` file, I need to call `states.filterByDone()`  twice. One for all non-done states and another for only done states (used for the `Done < 30 Days` page).

Reseting the collection makes the only done statuses unaccessible, as they're cleared when `states.filterByDone()` is first called.

Reply:
I think this could probably `this.clone().reset(this.filter`

It shouldn't just `new` a vanilla collection as it won't be a `BaseCollection` or a `states` collection.  In this case `new Collection` is the States Collection.

However since it is the case that we're calling this twice once `true` once `false` I think maybe we should change this to `groupByDone` and do something like:

```js
const { done, notDone } = this.groupBy(state => {
  return state.isDone() ? 'done' : 'notDone';
});

return {
  done: new Collection(done),
  notDone: new Collection(notDone),
};
```
Not sure if there's a better term for `notDone`

Reply:
Great way to do it. Updated to this in my last push.